Improving proposals limits on summary and title

andreslucena opened the pull request.

andreslucena committed:
Improving proposals limit to 1000 chars on summary and 160 chars on title

andreslucena committed:
Fixing tests on improving proposals limit

josepjaume commented on db/migrate/20160217115259_change_limit_on_title_proposal.rb:
```diff
@@ -0,0 +1,8 @@
+class ChangeLimitOnTitleProposal < ActiveRecord::Migration
+  def up
+    change_column :proposals, :title, :string, :limit => 161
```
Why 161 instead of 160? xD

josepjaume commented on lib/proposal_xls_importer.rb:
```diff
@@ -13,7 +13,9 @@ def import
-          create_proposal(proposal_data.to_attributes)
+          proposal = create_proposal(proposal_data.to_attributes)
+          result = proposal.save
+          debugger unless result
```
Careful :)

josepjaume commented on lib/proposal_xls_importer.rb:
```diff
@@ -13,7 +13,9 @@ def import
           proposal_data.parse!
-          create_proposal(proposal_data.to_attributes)
+          proposal = create_proposal(proposal_data.to_attributes)
+          result = proposal.save
```
Use `save!` here

josepjaume commented:
Do you think we could improve the seeds so proposals have longer summaries? I can then work a bit on the CSS.

andreslucena committed:
Proposals limits: including feedback from github

andreslucena committed:
Adding seeds to proposals limits

andreslucena commented:
I've just added the seeds. I'm having a problem on my local machine with the title 
  title: Faker::Lorem.paragraph.truncate(150) 

I get an error on rake db:seed, that title it's too long (!!). Probably it's something that I'm missing.

josepjaume committed:
Fix bug with measurable

By storing variables in class instance variables, the module itself was caching them, so the first value cached was used for validation, in this case, `Debates`. Funny thing is it didn't happen in development, as debates are feature-flagged.

josepjaume commented:
@andreslucena I've found the bug :D. It turns out that `limit` magic inside of `Measurable` was actually caching class variables in the `Measurable` module instead of in the inherited class. That made `Debates` take presedence *in the seeds* as it was called *first*. Tricky one.

Clever code is dumb code.

josepjaume committed:
Fix max proposal length

andreslucena commented:
:+1:
